### ISSUE TITLE
Failed subshells should prevent parent comments to run

### ISSUE THREAD

**Issue by lmorg** [lmorg/murex#407]
eg
```
ls @{g: blah blah}
```

Should fail in the same way that
```
ls $BLAHBLAH
```
might (where BLAHBLAH is unset)

**Comment by lmorg** [comment]
```
ls @{g: blah blah}
```
Currently fails (correctly) because of https://github.com/lmorg/murex/issues/409

However the following should fail:
```
set false=${false}
```

**Comment by lmorg** [comment]
Added to `develop` -- will be released in v2.7
